lib/model/gamers.js

```diff
@@ -1,31 +1,46 @@
-import apiLib from 'call-of-duty-api';
-import {insertIntoDatabase, queryDatabase, updateDatabaseRecords} from "../etl/utils";
-import WarzoneMapper from "../etl/mapper";
-import {GAMER_TABLE} from "../constants";
+import CallOfDutyApi from 'call-of-duty-api';
+
+import {GAMER_TABLE} from '../constants';
+
+import {insertIntoDatabase, queryDatabase, updateDatabaseRecords} from '../etl/utils';
+import WarzoneMapper from '../etl/mapper';
 import {login} from '../api_wrapper';
-const API = apiLib({platform: "battle", ratelimit: {maxRequests: 2, perMilliseconds: 1000, maxRPS: 2}});
+
+import UtilityService from './../services/UtilityService.js';
+
+const API = CallOfDutyApi({platform: 'battle', ratelimit: {maxRequests: 2, perMilliseconds: 1000, maxRPS: 2}});
+
 let EMAIL = process.env.WARZONE_EMAIL;
 let PASSWORD = process.env.WARZONE_PASSWORD;
+
+//===---==--=-=--==---===----===---==--=-=--==---===----//
+
+
+
 export function getGamerFromAPI(username, platform) {
-    return API.login(EMAIL, PASSWORD).then(() => {
-        return API.MWwz(username, platform);
-    });
+	return API.login(EMAIL, PASSWORD).then(() => {
+		return API.MWwz(username, platform);
+	});
 }
 
+
+
 export async function initializeGamer(queryGamer) {
-    let API = await login();
-    let gamer = await API.MWwz(queryGamer.username, queryGamer.platform);
-    gamer.needs_backfill = true;
-    return insertIntoDatabase(WarzoneMapper.mapGamer(gamer), GAMER_TABLE);
+	let API = await login();
+	let gamer = await API.MWwz(queryGamer.username, queryGamer.platform);
+	gamer.needs_backfill = true;
+	return insertIntoDatabase(WarzoneMapper.mapGamer(gamer), GAMER_TABLE);
 }
 
-export function updateGamer(query, gamer){
-    return updateDatabaseRecords(query, gamer, GAMER_TABLE);
+
+
+export function updateGamer(query, gamer) {
+	return updateDatabaseRecords(query, gamer, GAMER_TABLE);
 }
 
-export function queryGamers(query = null){
-    if(!query){
-        query = {}
-    }
-    return queryDatabase(GAMER_TABLE, query)
+
+
+export function queryGamers(query) {
+	query = UtilityService.validateItem(query, 'object', {});
```

Comment:
The thing I use this most for is this example here, if query was an array, function, string, etc, it would be okay and go to the next step and probably throw an error. validateItem makes sure it's always the right type

Reply:
I'm wondering if we should move everything to TypeScript so we get these checks at compile time instead of runtime?

Reply:
I've never used typescript beyond a documentation purpose, but I'm down for whatever. This is a flexible library and when we switch to typescript, this will be easy to remove i'm sure.